### DISCUSSION TITLE
Device name when advertising

### DISCUSSION THREAD

**Discussion by mpanzica** [shinyorg/shiny#1125] [Q&A]
Hello,
I have an application running as BLE host. Using any BLE scanner I can see the list of advertised services, but the device name is blank (N/A) and it can only be retrieved by connecting and then reading 1800/2A00. 
Is there any way to make the device name being visible during scan?

Thanks.
M.

**Comment by aritchie** [comment]
You can set the local name in the advertisement options that you pass to StartAdvertising

**Reply by mpanzica** [reply]
That is exactly what I do. Infact I can read the string I set on LocalNames by querying 1800/2A00. 
Still I don't understand why I can't see the name while scanning, even with BLE Scanner or nRF Connect neither on iOS nor on Android.
Am I missing anything?
Oh, the app is running on Android.

**Reply by aritchie** [reply]
Please link a github repository.  I don't have anywhere near enough info to guess.  

Permissions, version of android, local name is too big, are you getting anything in the android logs when debugging, etc.

**Comment by mpanzica** [comment]
Hi Allan,

Sorry for being so late.
Please find a test project at the following url: 

https://github.com/mpanzica/BleTest

These are the issues I am experiencing:
(o) Despite the Local name being set in AdvertisementOptions, it doesn't seem to appear while scanning (tools like BLE Scanner and nRF Connect report N/A).
(o) Setting the LocalName in AdvertisementOptions changes my BLE device name permanently (I have to change it back from BT settings)
(o) The 2 second notification (excerpt from your samples) doesn't seem to work.
(o) I can see notification subscribers but couldn't manage to notify them with a test code.
What am I doing wrong? Any help would be pretty much appreciated.

Thank you in advance for your support.
Kindest regards,
M.

PS: Tested on Pixel 6 with Android 13 and latest patches applied.

**Reply by aritchie** [reply]
> (o) Despite the Local name being set in AdvertisementOptions, it doesn't seem to appear while scanning (tools like BLE Scanner and nRF Connect report N/A). Setting the LocalName in AdvertisementOptions changes my BLE device name permanently (I have to change it back from BT settings)

I set the local name and the device name.  This is what was requested and my use-case.  If scanners aren't looking at local name, I can't change that.  The fact that the device name is changing is just part of the feature set.

> The 2 second notification (excerpt from your samples) doesn't seem to work.

Add a .Switch().  RX Observables do not automatically execute as it is only a select. 


> Thank you in advance for your support.

This is not meant as a tech support area as this is free source code.  Please be respectful of maintainer time.

**Comment by mpanzica** [comment]
Sorry Allan, didn't mean to be disrespectful of your time!
I have used your samples as a starting point just for not bother you. 
I can't see "Switch()" in your BleHosting sample where the Observable portion of code is copied from. 
That's why I've thought there must have been something wrong either with the sample or with my code.

**Reply by mpanzica** [reply]
Glad to see that you have added .Switch() to your BluetoothLE-Hosting sample.
Still the sample doesn't compile at all.

Kindest regards,
M.

**Reply by aritchie** [reply]
It's because I don't keep every sample up to date and that one is quite old.  Samples are samples, treat them as such.  I'll update them as I have time.
